site/cypress/e2e/Cypress-Feature/grade_inquiries.spec.js

```diff
@@ -0,0 +1,47 @@
+describe('Test cases revolving around grade inquires', () => {
+    const setGradeInquiriesForGradeable = (gradeableId, date = null) => {
+        cy.visit(['sample', 'gradeable', gradeableId, 'update']);
+        cy.get('[data-testid="yes-grade-inquiry-allowed"]').click();
+        cy.contains('Dates').click();
+        cy.get('[data-testid="date-grade-inquiry-due"]').click();
+        cy.get('[data-testid="date-grade-inquiry-due"]').should('be.visible');
+        cy.get('[data-testid="date-grade-inquiry-due"]').clear().type(date, { parseSpecialCharSequences: false, force: true });
+    };
+
+    it('should test normal submission grade inquiry panel', () => {
+        cy.login();
+        const gradeableId = 'grades_released_homework';
+        const gradeInquiryDeadlineDate = '9998-01-01 00:00:00';
+        setGradeInquiriesForGradeable(gradeableId, gradeInquiryDeadlineDate);
+        cy.visit(['sample', 'gradeable', gradeableId, 'grading', 'details']);
+        cy.get('[data-testid="view-sections"]').click();
+        cy.get('[data-testid="grade-button"]').eq(3).click();
+        cy.get('[data-testid="grade-inquiry-info-btn"]').click();
+        cy.get('[data-testid="grading-label"]').should('contain', 'Grade Inquiry');
+        cy.get('[data-testid="grade-inquiry-actions"]').contains('Submit Grade Inquiry').and('be.disabled');
+        cy.get('[data-testid="reply-text-area-0"]').click().type('Submitty');
+        cy.get('[data-testid="markdown-mode-tab-preview"]').first().should('exist');
+        cy.get('[data-testid="grade-inquiry-actions"]').contains('Submit Grade Inquiry').and('not.be.disabled');
+    });
+    ['ta', 'grader'].forEach((user) => {
+        it(`${user} can see grade inquiry panel`, () => {
+            cy.login(user);
+            const gradeableId = 'grades_released_homework';
+            cy.visit(['sample', 'gradeable', gradeableId, 'grading', 'details']);
+            cy.get('[data-testid="popup-window"]').should('exist');
+            cy.get('[data-testid="close-button"]').should('exist');
+            cy.get('[data-testid="close-hidden-button"]').should('exist');
+            cy.get('[data-testid="agree-popup-btn"]').click();
+            if ( user === 'ta') {
+                cy.get('[data-testid="view-sections"]').click();
+            }
+            cy.get('[data-testid="grade-button"]').eq(3).click();
+            cy.get('[data-testid="grade-inquiry-info-btn"]').click();
+            cy.get('[data-testid="grading-label"]').should('contain', 'Grade Inquiry');
+            cy.get('[data-testid="grade-inquiry-actions"]').contains('Submit Grade Inquiry').and('be.disabled');
```

Comment:
```suggestion
            cy.get('[data-testid="grade-inquiry-actions"]').should('contain', 'Submit Grade Inquiry').and('be.disabled');
```

Apply for other assertions of contains

Reply:
here i am searching for it , previously i have done in that manner with some other data-testid but it failing for that so i change (what i think that element is disable so select just one top level then use `.contains `to search and then verify it ) but its failing again 

should i change to that one ??

Reply:
If you are getting the action that contains 'submit grade inquiry', that is okay, but then change the assertion to ```contains(xxx).should('be.disabled')```